### ISSUE TITLE
[css-shapes] Remove padding-box, content-box values from <shape-box>.

### ISSUE THREAD

**Issue by bfgeek** [w3c/csswg-drafts#3872]
https://drafts.csswg.org/css-shapes-1/#shapes-from-box-values

padding-box, content-box aren't general enough to fix within the wider CSS box model. E.g. consider an element with scrollable overflow:
https://www.software.hixie.ch/utilities/js/live-dom-viewer/?saved=6857

In the above example the content-box of the element with shape-outside is technically larger than the element itself, but the specification doesn't (? i believe) mention how do deal with this.

Last year we also added use-counters within our codebase for how often these values were getting used:
padding-box:
https://www.chromestatus.com/metrics/feature/timeline/popularity/2426
(when making the value non-parsable, sites have minimal (<2px) behaviour change).

content-box:
https://www.chromestatus.com/metrics/feature/timeline/popularity/2425
The use-counter for above was over-reporting due to a bug in the code, this was now fixed, and most reports are for a buggy version where it considered images as using content-box.
I couldn't find any sites actually using this value, and relying on its behaviour.

border-box:
https://www.chromestatus.com/metrics/feature/timeline/popularity/2427
There are a couple of sites which would have a large behaviour change if we'd remove this. However they'd have the same behaviour as if they had shape-outside:margin-box.
Unlike the other values border-box does have a use-case for web developers.

Given the values don't generalize to the wider css box-model, the zero use-counts for this feature, I believe that we can simplify this feature, and remove these values.

I'm not proposing removing border-box, however this could be considered for simplicity. border-box *does* have a relatively strong use-case compared to the other values.

**Comment by bfgeek** [comment]
cc/ @astearns , @tabatkins

**Comment by rolivo** [comment]
Why :(, This module is beautiful...

**Comment by tabatkins** [comment]
> padding-box, content-box aren't general enough to fix within the wider CSS box model. E.g. consider an element with scrollable overflow:
> https://www.software.hixie.ch/utilities/js/live-dom-viewer/?saved=6857
>
> In the above example the content-box of the element with shape-outside is technically larger than the element itself, but the specification doesn't (? i believe) mention how do deal with this.

I'm not sure what you mean by this. padding-box is well-defined, and is not the same as the scrollable overflow, sure. So the result you get is well-defined, if a little weird.

That said, I agree the use-cases for the lesser boxes seem pretty minimal; I think we just reused the box definitions because they were there.

> There are a couple of sites which would have a large behaviour change if we'd remove this. However they'd have the same behaviour as if they had shape-outside:margin-box.

What do you mean by this? That switching to margin-box would cause a large behavior change for them?

**Comment by bfgeek** [comment]
> I'm not sure what you mean by this. padding-box is well-defined, and is not the same as the scrollable overflow, sure. So the result you get is well-defined, if a little weird.

Sure - well-defined - that results in bad, strange behaviour ;) .

> What do you mean by this? That switching to margin-box would cause a large behavior change for them?

Some of the sites have:
```css
width: 50px;
height: 50px;
border-radius: 25px;
shape-outside: border-box;
```

To create a circle shape area.
Removing the `border-box` value would mean that they'd lose that behaviour. However if they notice the broken behaviour they could replace their CSS with:
```css
width: 50px;
height: 50px;
border-radius: 25px;
shape-outside: margin-box;
```

And it would work fine.

**Comment by tabatkins** [comment]
Ah, k, so removing it *would* break them, even if it's fixable.

I think we probably shouldn't remove border-box, then. ^_^  It has a reasonable usage beyond margin-box, I think; using the box's margin to push it away from the edge, but having text wrap a *different* distance from it, seems totally reasonable.

**Comment by aethanyc** [comment]
@heycam points to me this discussion. As a firefox shapes implementer, I'm OK with removing `padding-box` and `content-box` to simplify code. Note that shape-box can be used with basic-shape. I'm not sure the use-counter include the usage with shape-box and basic-shape combined. 

cc @jensimmons

**Comment by AmeliaBR** [comment]
I would be very disappointed if we just removed these values, as that would prevent authors from aligning shapes with actual content elements or with content or padding-clipped background.

(Note that the `<shape-box>` value has two effects: it can define the shape directly, OR it can act as a reference box for aligning and scaling shapes defined by images or the shape functions.)

But, it is clear that we have unspecified behavior. We need rules for what happens in scrolling containers. Maybe what should happen is that shapes aligned to these boxes instead gets treated as border box at used value time if the element has scrollable overflow.

Regarding the use counter: current usage of `shape-outside` is often limited by support for `clip-path`. A [`clip-path` is also supposed to support many reference boxes](https://drafts.fxtf.org/css-masking-1/#the-clip-path), but Chrome does not support it. Chrome only supports the default `border-box`, so if you want a `shape-outside` shape to match a `clip-path` shape, you currently need to make your `shape-outside` shape reference `border-box`.

Of course, this also means that any decisions about reference boxes at scrollable elements should affect `clip-path`, too. Which means, they should also probably affect [`mask-origin`](https://drafts.fxtf.org/css-masking-1/#the-mask-origin), which defines reference boxes for masked elements. But `mask-origin` is intentionally designed to be consistent with `background-origin`, and backgrounds in scrollable containers apply to the inside dimensions, not the outside dimensions that are used for the float wrapping area, which is what `shape-outside` relates to…

In other words, we're not going to define a rule that is consistent everywhere when it comes to scrollable elements. But that doesn't mean that we should prevent authors from using these properties in a consistent way on _non_ scrolling elements.
